scripts/system_performance/controller/method_mapping.py

```diff
@@ -0,0 +1,218 @@
+        return instrument_list
+
+    def run_every_instrument(self, instrument_dict, method_name, method_arguments=None):
+        """Returns all existing instruments in a dictionary for a given method as:
+        {instrument_name : [[method output row 1]],[[method output row 2] etc] ...}"""
+
+        for instrument in self.get_instrument_models():
+            try:
+                # if method_name in self.create_method_mappings():
+                method_arguments['instrument_id'] = int(instrument.id)
+                logging.info("Querying for instrument: {}".format(method_arguments))
+                instrument_dict[instrument[1]] = self.create_method_mappings()[method_name](**method_arguments)
+            except KeyError:
+                logging.warn("Invalid Input - method '%s' does not exist try -help "
+                                 "to look at existing methods and arguments", method_name)
+        return instrument_dict
+
+    def user_instrument_list_validate(self, instrument_input):
+        """compares list of instruments from user input with instruments in db.
+        :return tuple of valid instruments in user input"""
+
+        valid_instruments_list = []
+        list_of_instruments_from_db = self.get_instrument_models()
+
+        for instrument_from_db in list_of_instruments_from_db:
+
+            for instrument_from_user in instrument_input:
+                if instrument_from_db[1] == instrument_from_user:
+                    valid_instruments_list.append(instrument_from_db)
+
+        return valid_instruments_list
+
+    def get_query_for_instruments(self, method_name, instrument_input=None, additional_method_arguments=None):
+        """Checks that the instrument_from_db's in method input exist and then calls
+        and returns methods specified as input for each instrument_from_db placing in
+        a dictionary as a nested list for each instrument_from_db as:
+         {instrument_name : [[method output row 1]],[[method output row 2] etc.] ...}
+
+        Run methods for each instrument_from_db name
+        :param instrument_input: [(str), (str), ...] - Represents the instruments to be used in queries
+        :param method_name: (str) -  The name of the methods to run
+        :param additional_method_arguments - method arguments specified by user
+        :return: A dictionary of instrument_from_db names (key) and list of method output (value)
+         """
+
+        instrument_dict = {}
+
+        # To allow for keys to still be added to dictionary such as the instrument_id
+        if not additional_method_arguments:
+            additional_method_arguments = {}
+
+        # If input is None, run all instruments by default
+        if not isinstance(instrument_input, list):
+            logging.error("Value is not iterable, the first argument must be of type list")
+            return None
+
+        for instrument in instrument_input:
```

Comment:
might be able to compress the complexity of this. If you do something like the following:
```
if instrument == 'all':
...
else:
   valid_instruments = self.user_instrument_list_validate(instrument_input)
   for instruments in valid_instruments:
       ...
```
Let me know if that doesn't make any sense - but I think it is possible to do given the below logic